src/sorcha/modules/PPLinkingFilter.py

```diff
@@ -49,105 +48,50 @@ def PPLinkingFilter(
-
-        separation_df = pd.DataFrame({"night": obs_above_min["night"].unique(), "separation": separation})
-
-        # merge on and drop all observations where separation < minimum separation
-        obs_sep = pd.merge(obs_above_min, separation_df, on="night")
-        obs_above_min_sep = obs_sep.loc[obs_sep["separation"] >= minimum_separation]
-        obs_above_min_sep.reset_index(inplace=True, drop=True)
-
-        # quick check: if number of nights left is less than min_tracklets, skip this object
-        # as there aren't enough tracklets to count as a full track
-        if len(obs_above_min_sep["night"].unique()) < min_tracklets:
-            continue
-
-        unique_nights = obs_above_min_sep["night"].unique()
-
-        # checks to see if a track containing a number of tracklets >= min_tracklets
-        # exists in a window of days <= tracklet_interval.
-        # assumes all observations in a night are part of a valid tracklet
-        # which should be true at this point
-
-        # once a linking is made, the night on which the linking was made is stored
-        linked_night = False
-        for i, night in enumerate(unique_nights):
-            if i + min_tracklets - 1 >= len(unique_nights):
-                break
-
-            diff = unique_nights[i + min_tracklets - 1] - night
-
-            if diff < tracklet_interval:
-                linked_night = night
-                break
-
-        # get all observations of this object made on or after the night it was linked
-        if linked_night:
-            obs_final_night = obs_object[obs_object["night"] >= linked_night]
+    # create a small dataframe for difi to work on with only the relevant columns
+
+    difi_dataframe = pd.DataFrame(
+        {
+            "object_id": observations["ObjID"].astype(object),
+            "obs_id": observations["FieldID"],
+            "time": observations["FieldMJD"],
+            "night": observations["night"],
+            "ra": observations["AstRA(deg)"],
+            "dec": observations["AstDec(deg)"],
+        }
+    )
 
-            # get the original index numbers of the observations that made it through
-            # append them to the final index
-            final_index = obs_final_night["original_index"].values
-            final_idx.append(final_index.tolist())
+    metric = NightlyLinkagesMetric(
+        linkage_min_obs=min_observations,  # min observations per tracklet
+        max_obs_separation=maximum_time,  # maximum time separation between subsequent observations in a tracklet [days]
+        min_linkage_nights=min_tracklets,  # minimum number of nights on which a tracklet must be detected
+        min_obs_angular_separation=minimum_separation,  # minimum angular separation between observations in a tracklet [arcseconds]
+    )
 
-    # flatten the final index
-    flat_idx = [x for xs in final_idx for x in xs]
+    findable, window_summary = metric.run(
```

Comment:
Minor point: I don't _think_ that we're using `window_summary` after this. If that's true then we can replace it with `_`, i.e. `findable, _ = metric.run(...)`, to avoid creating the new variable. But if `window_summary` is, or will be, used, then please ignore this comment. :)